install-template.sh

```diff
@@ -40,13 +68,31 @@ else
     exit 1
   fi
   install_nix
+  test_nix_installation || (echo "Failed to install nix, cancelling installation." && exit 1)
   echo nix is now installed:
   nix --version
 fi
-test_nix_installation
 
-echo TODO: testing binary cache...
-# is there a good way to test whether a binary cache is configured and available?
+echo
+echo BINARY CACHE CONFIGURATION
+echo ==========================
+
+if test_cache_configured; then
+  echo Hooray, the garnix cache is configured.
+else
+  echo For \'@@toolName@@\' to work well, it is recommended that you add the
+  echo garnix.io binary cache to your nix configuration. But the cache
+  echo configuration cannot be found.
+  printf "Should I add the garnix.io binary cache to /etc/nix/nix.conf now (requires sudo)? [y/n] "
```

Comment:
Why sometimes echo sometimes printf?

Reply:
`printf` doesn't print a trailing newline. So for questions it's `printf`. (I first used `echo -n`, but `shellcheck` says that that's not portable.)